### PR TITLE
[Log View Bug] Bug fix: log view scroll

### PR DESCRIPTION
**Pull request by mlunoe**
---
~~⚠️ Dependent on this branch: https://github.com/dcos/dcos-ui/pull/358~~

~~ℹ️ REVIEWER LOOK HERE (diff between this and its depending branch): https://github.com/dcos/dcos-ui/compare/mlunoe/make-task-detail-tabs-routable...mlunoe/bug-fix-log-view-scroll?expand=1~~

---
Node Detail Page log tab scrolls within log view:

Scrolls whole page on any other nodes detail page

Service Detail Page log tab scrolls within log view:

Scrolls whole page on any other service detail page

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/js/pages/services/task-details/TaskDetailsTab.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import Units from '../../../utils/Units';
+
+class TaskDetailsTab extends React.Component {
+  getMesosTaskDetailsDescriptionList(mesosTask) {
+    if (mesosTask == null || !MesosSummaryStore.get('statesProcessed')) {
+      return null;
+    }
+
+    let headerValueMapping = {'ID': mesosTask.id};
+
```
Comment (MatApple) [inline src/js/pages/services/task-details/TaskDetailsTab.js]: #nit don't need the extra line

File: src/js/pages/services/task-details/TaskDetailsTab.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    let node = MesosStateStore.getNodeFromID(mesosTask.slave_id);
+
+    if (node != null) {
+      headerValueMapping['Node'] = `${node.hostname} (${node.id})`;
+    }
+
+    let sandBoxPath = TaskDirectoryStore.get('sandBoxPath');
+    if (sandBoxPath) {
```
Comment (MatApple) [inline src/js/pages/services/task-details/TaskDetailsTab.js]: Sometimes have whitespace before/after a `let` or `if` statement in this method. Maybe you could pick one way and make sure spacing is consistent in the method :)

File: src/js/pages/services/task-details/TaskDetailsTab.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      }
+
+      let colorIndex = resourceColors[resource];
+      let resourceLabel = resourceLabels[resource];
+      let resourceIconClasses = `icon icon-sprite icon-sprite-medium
+        icon-sprite-medium-color icon-resources-${resourceLabel.toLowerCase()}`;
+      let resourceValue = Units.formatResource(
+        resource, task.resources[resource]
+      );
+      return (
```
Comment (MatApple) [inline src/js/pages/services/task-details/TaskDetailsTab.js]: whitespace above this line :)

File: src/js/pages/services/task-details/__tests__/TaskDetail-test.js [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
+jest.dontMock('../../../../components/SidePanelContents');
+jest.dontMock('../TaskFilesTab');
 jest.dontMock('../TaskDetail');
-jest.dontMock('../../stores/MesosStateStore');
-jest.dontMock('../../mixins/GetSetMixin');
+jest.dontMock('../../../../stores/MesosStateStore');
+jest.dontMock('../../../../mixins/GetSetMixin');
 
-var JestUtil = require('../../utils/JestUtil');
+var JestUtil = require('../../../../utils/JestUtil');
```
Comment (MatApple) [inline src/js/pages/services/task-details/__tests__/TaskDetail-test.js]: Lets change these vars to lets in this file!

File: src/js/pages/services/task-details/__tests__/TaskLogsTab-test.js [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
-jest.dontMock('../TaskDebugView');
-jest.dontMock('../FilterBar');
+jest.dontMock('../../../../components/icons/IconDownload');
+jest.dontMock('../../../../components/MesosLogView');
+jest.dontMock('../../../../components/RequestErrorMsg');
+jest.dontMock('../TaskLogsTab');
+jest.dontMock('../../../../components/FilterBar');
 
-var JestUtil = require('../../utils/JestUtil');
+var JestUtil = require('../../../../utils/JestUtil');
```
Comment (MatApple) [inline src/js/pages/services/task-details/__tests__/TaskLogsTab-test.js]: `let`

File: src/js/routes/nodes.js [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
 
 import HostTable from '../components/HostTable';
 import NodesPage from '../pages/NodesPage';
 import NodesGridView from '../components/NodesGridView';
 import NodeDetailPage from '../pages/nodes/NodeDetailPage';
-import TaskDetail from '../components/TaskDetail';
+import TaskDetail from '../pages/services/task-details/TaskDetail';
+import TaskFilesTab from '../pages/services/task-details/TaskFilesTab';
+import TaskLogsTab from '../pages/services/task-details/TaskLogsTab';
+import TaskDetailsTab from '../pages/services/task-details/TaskDetailsTab';
```
Comment (MatApple) [inline src/js/routes/nodes.js]: 🔠

File: src/js/routes/nodes.js [hunk 7]
```diff
@@ -45,15 +46,35 @@ let nodesRoutes = {
...
         {
           type: Route,
+          path: 'tasks/:taskID/?',
           name: 'nodes-task-details',
-          path: 'task-detail/:taskID/?',
-          handler: TaskDetail
+          handler: TaskDetail,
+          children: [
+            {
+              type: DefaultRoute,
```
Comment (MatApple) [inline src/js/routes/nodes.js]: Nice, we should use this more often

File: src/js/pages/services/task-details/TaskDetailsTab.js (hunk 1 above)
Comment (mlunoe) [inline src/js/pages/services/task-details/TaskDetailsTab.js]: done

File: src/js/pages/services/task-details/TaskDetailsTab.js (hunk 2 above)
Comment (mlunoe) [inline src/js/pages/services/task-details/TaskDetailsTab.js]: done

File: src/js/pages/services/task-details/TaskDetailsTab.js (hunk 3 above)
Comment (mlunoe) [inline src/js/pages/services/task-details/TaskDetailsTab.js]: done

File: src/js/pages/services/task-details/__tests__/TaskDetail-test.js (hunk 4 above)
Comment (mlunoe) [inline src/js/pages/services/task-details/__tests__/TaskDetail-test.js]: done

File: src/js/pages/services/task-details/__tests__/TaskLogsTab-test.js (hunk 5 above)
Comment (mlunoe) [inline src/js/pages/services/task-details/__tests__/TaskLogsTab-test.js]: done

File: src/js/routes/nodes.js (hunk 6 above)
Comment (mlunoe) [inline src/js/routes/nodes.js]: done

**Comment by mlunoe** [comment]
Comments addressed and pushed to https://github.com/dcos/dcos-ui/pull/358

**Comment by jfurrow** [comment]
As I said in the wrong PR, I would like to be able to set a minimum height on the log viewer to account for short viewports. I'm trying to throw this together locally, I'll update this when I've got it working properly.

File: src/js/pages/services/ServicesTab.js [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import classNames from 'classnames';
```
Comment (kennyt) [inline src/js/pages/services/ServicesTab.js]: :abc:

File: src/js/pages/ServicesPage.js [hunk 9]
```diff
@@ -53,9 +53,13 @@ var ServicesPage = React.createClass({
   },
 
   render: function () {
+    // Make sure to grow when logs are displayed
+    let routes = this.context.router.getCurrentRoutes();
+
     return (
       <Page
         navigation={this.getNavigation()}
+        dontScroll={routes[routes.length - 1].dontScroll}
```
Comment (kennyt) [inline src/js/pages/ServicesPage.js]: i wonder if we should create a method on `RouteUtil` called `getLastRoute` and only needs the router

File: src/js/components/Page.js [hunk 10]
```diff
@@ -95,20 +97,54 @@ var Page = React.createClass({
...
+        className={contentClassSet}
+        ref="gemini">
+        {content}
+      </GeminiScrollbar>
+    );
+  },
+
   render: function () {
-    let {className, navigation, title} = this.props;
+    let {className, navigation, dontScroll, title} = this.props;
```
Comment (kennyt) [inline src/js/components/Page.js]: what do you think about adding some propTypes for this component?

**Comment by jfurrow** [comment]
I was just playing with this locally, and I think we need to find a way to give the log viewer a minimum height... though it's more involved than I expected, so let's just tackle it another time. Here's a JIRA! https://mesosphere.atlassian.net/browse/DCOS-7684

**Comment by rcorral** [comment]
flicts!

File: src/js/pages/ServicesPage.js (hunk 9 above)
Comment (mlunoe) [inline src/js/pages/ServicesPage.js]: I would really like to create a task to get that done outside of this PR. This PR has already dragged along for some time now and it doesn't feel like a part of this task. I created a task for it

File: src/js/components/Page.js (hunk 10 above)
Comment (mlunoe) [inline src/js/components/Page.js]: Done

**Comment by mlunoe** [comment]
All comments have been addressed. ping @daiyiwhale @jfurrow

**Comment by daiyiwhale** [comment]
👍

**Comment by daiyiwhale** [comment]
Oh wait, scroll not working with `Show scroll bars: automatically based on mouse or trackpad` ):

**Pull request merged**